src/test/components/MenuButtons.test.js

```diff
@@ -14,60 +14,210 @@ import {
+
+    afterAll(async function() {
+      delete (window: any).TextEncoder;
+    });
+
+    beforeEach(function() {
+      // Flow doesn't know uploadBinaryProfileData is a jest mock.
+      (uploadBinaryProfileData: any).mockImplementation(() =>
+        Promise.resolve(hash)
+      );
+      // Flow doesn't know sha1 is a jest mock.
+      (sha1: any).mockImplementation((_data: Uint8Array) =>
+        Promise.resolve(hash)
+      );
+      // Flow doesn't know shortenUrl is a jest mock.
+      (shortenUrl: any).mockImplementation(() =>
+        Promise.resolve('https://perf-html.io/')
+      );
+    });
+
+    // profile.meta.networkURLsRemoved flag is set to false as a default.
+    it('render correctly', () => {
+      const { store, container } = setup();
+      store.dispatch(startSymbolicating());
+
+      // MenuButtons is rendering a fragment with several children. We need to
+      // check all children to assess that the component renders properly.
+      expect(Array.from(container.children)).toMatchSnapshot();
+
+      store.dispatch(doneSymbolicating());
+      expect(Array.from(container.children)).toMatchSnapshot();
+    });
+
+    it('render buttons with profile.meta.networkURLsRemoved set to true', () => {
+      const profile = processProfile(createGeckoProfileWithJsTimings());
+      profile.meta.networkURLsRemoved = true;
+      const { store, container } = setup(profile);
+      store.dispatch(startSymbolicating());
+
+      expect(Array.from(container.children)).toMatchSnapshot();
+
+      store.dispatch(doneSymbolicating());
+      expect(Array.from(container.children)).toMatchSnapshot();
+    });
+
+    it('render buttons with profile.meta.networkURLsRemoved set to undefined', () => {
+      const profile = processProfile(createGeckoProfileWithJsTimings());
+      profile.meta.networkURLsRemoved = undefined;
+      const { store, container } = setup(profile);
+      store.dispatch(startSymbolicating());
+
+      expect(Array.from(container.children)).toMatchSnapshot();
+
+      store.dispatch(doneSymbolicating());
+      expect(Array.from(container.children)).toMatchSnapshot();
+    });
+
+    it('render the Privacy Notice panel when Share button is clicked', () => {
+      const { store, container, getShareButton } = setup();
+      store.dispatch(doneSymbolicating());
+      fireEvent.click(getShareButton());
+
+      expect(Array.from(container.children)).toMatchSnapshot();
+    });
+
+    it('should share the profile when Share button is clicked', async () => {
+      const {
+        store,
+        //container,
+        getShareButton,
+        getShareWithUrlsButton,
+        getInnerShareButton,
+        getInnerShareWithUrlsButton,
+        getMenuButtonsContainer,
+        //getPermalinkButton,
+      } = setup();
+      // Sharing without URLs
+      fireEvent.click(getShareButton());
+      fireEvent.click(getInnerShareButton());
+      await waitForDomChange({ container: getMenuButtonsContainer() }).then(
```

Comment:
I couldn't use `waitForElement` here as we talked on slack. It only expects an element to appear or dissapear but we only change some css values(transition) of the input. So I checked the container to wait for css class changes. It adds `.currentButtonIsPermalinkButton` when profile is uploaded.

Reply:
Can you add some code comments explaining the compromise here? This is hooking a bit into the implementation logic of the component. It's good enough for now, but it would be nice to include a code comment about it.